In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from sklearn.ensemble import RandomForestClassifier
from sklearn import preprocessing
from sklearn.model_selection import train_test_split 
# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/titanic/train.csv
/kaggle/input/titanic/test.csv
/kaggle/input/titanic/gender_submission.csv


This is the import section and this shows the directory of each file (its path)
Steps: import all the above needed libraries
Conclusion: As shown the files needed to work on train and test files and gender submission that has the accurate values of the whole problem.

In [2]:
train_data = pd.read_csv("/kaggle/input/titanic/train.csv")
train_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

This reads the data inside train.csv file and put it in train_data then show it by train_data.head()
Steps: reading data in file then showing samples of existing data by train_data.head()
conclusion: Shows samples of the columns inside train.csv file.

In [3]:
train_data.info()
print(train_data.columns)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')


Here i wanted to see the info to see that not all the features are 891 and this means that age and Embarked and cabin has null values. Steps: writing the the info inside train file the printing the columns with number of details inside each column. Conclusion: showing 714 in age and 204 in cabin and 889 in Embarked,this shows that they contains many null.

In [4]:
cols = ['Name', 'Ticket', 'Cabin']
train_data = train_data.drop(cols, axis=1)
train_data.info()
train_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Sex          891 non-null    object 
 4   Age          714 non-null    float64
 5   SibSp        891 non-null    int64  
 6   Parch        891 non-null    int64  
 7   Fare         891 non-null    float64
 8   Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(2)
memory usage: 62.8+ KB


PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0            1         0       3    male  22.0      1      0   7.2500        S
1            2         1       1  female  38.0      1      0  71.2833        C
2            3         1       3  female  26.0      0      0   7.9250        S
3            4         1       1  female  35.0      1      0  53.1000        S
4            5         0       3    male  35.0      0      0   8.0500        S

Here i dropped the features Name,ticket and cabin from train_data file as they will not be useful in my submission. Steps: Dtermining the names of columns needed to be dropped then drop them. Conclusion: Dropping the needed columns from the data file as shown above.

In [5]:
train_data['Age'] = train_data['Age'].interpolate()
train_data['Fare'] = train_data['Fare'].interpolate()
train_data['Fare'].fillna(train_data['Fare'],inplace= True)
train_data['Age'].fillna(train_data['Age'],inplace= True)
train_data['Embarked'].fillna("Unavailable",inplace= True)
print(train_data)

     PassengerId  Survived  Pclass     Sex   Age  SibSp  Parch     Fare  \
0              1         0       3    male  22.0      1      0   7.2500   
1              2         1       1  female  38.0      1      0  71.2833   
2              3         1       3  female  26.0      0      0   7.9250   
3              4         1       1  female  35.0      1      0  53.1000   
4              5         0       3    male  35.0      0      0   8.0500   
..           ...       ...     ...     ...   ...    ...    ...      ...   
886          887         0       2    male  27.0      0      0  13.0000   
887          888         1       1  female  19.0      0      0  30.0000   
888          889         0       3  female  22.5      1      2  23.4500   
889          890         1       1    male  26.0      0      0  30.0000   
890          891         0       3    male  32.0      0      0   7.7500   

    Embarked  
0          S  
1          C  
2          S  
3          S  
4          S  
..       

Here i make interpolation and fillna to Age and Fare to fill the missing values then i used fillna in Embarked to write in the NaN items unavailable. Steps: Interpolate both age and fare by interpolation method then fillna inside each one of them that fill the null values with the value of interpolation of each of them.

In [6]:
train_data['Sex'] = preprocessing.LabelEncoder().fit_transform(train_data['Sex'])
train_data['Embarked'] = preprocessing.LabelEncoder().fit_transform(train_data['Embarked'])
train_data[['Sex','Embarked']].info()
train_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Sex       891 non-null    int64
 1   Embarked  891 non-null    int64
dtypes: int64(2)
memory usage: 14.0 KB


PassengerId  Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0            1         0       3    1  22.0      1      0   7.2500         2
1            2         1       1    0  38.0      1      0  71.2833         0
2            3         1       3    0  26.0      0      0   7.9250         2
3            4         1       1    0  35.0      1      0  53.1000         2
4            5         0       3    1  35.0      0      0   8.0500         2

Here i converted all letters to numbers by using labelencoder.fittransform to sex and Embarked.
Steps:Using the method labelencoder to sex and Embarked. Conclusion: the letters in each are transformed to numbers as shown above.

In [7]:
test_data = pd.read_csv("/kaggle/input/titanic/test.csv")
test_data.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

Here i want to read the test file and the data inside it.
Steps: reading data in file then showing samples of existing data by test_data.head()
conclusion: Shows samples of the columns inside test.csv file.

In [8]:
cols = ['Name', 'Ticket', 'Cabin']
test_data = test_data.drop(cols, axis=1)
test_data.info()
test_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Pclass       418 non-null    int64  
 2   Sex          418 non-null    object 
 3   Age          332 non-null    float64
 4   SibSp        418 non-null    int64  
 5   Parch        418 non-null    int64  
 6   Fare         417 non-null    float64
 7   Embarked     418 non-null    object 
dtypes: float64(2), int64(4), object(2)
memory usage: 26.2+ KB


PassengerId  Pclass     Sex   Age  SibSp  Parch     Fare Embarked
0          892       3    male  34.5      0      0   7.8292        Q
1          893       3  female  47.0      1      0   7.0000        S
2          894       2    male  62.0      0      0   9.6875        Q
3          895       3    male  27.0      0      0   8.6625        S
4          896       3  female  22.0      1      1  12.2875        S

What we have made in train data file will be made here in test data file, i removed Name, ticket and cabin from the columns.
Steps: Dtermining the names of columns needed to be dropped then drop them. Conclusion: Dropping the needed columns from the data file as shown above.

In [9]:
test_data['Age'] = test_data['Age'].interpolate()
test_data['Fare'] = test_data['Fare'].interpolate()
test_data['Fare'].fillna(test_data['Fare'],inplace= True)
test_data['Age'].fillna(test_data['Age'],inplace= True)
test_data['Embarked'].fillna("Unavailable",inplace= True)
print(test_data)

     PassengerId  Pclass     Sex   Age  SibSp  Parch      Fare Embarked
0            892       3    male  34.5      0      0    7.8292        Q
1            893       3  female  47.0      1      0    7.0000        S
2            894       2    male  62.0      0      0    9.6875        Q
3            895       3    male  27.0      0      0    8.6625        S
4            896       3  female  22.0      1      1   12.2875        S
..           ...     ...     ...   ...    ...    ...       ...      ...
413         1305       3    male  33.5      0      0    8.0500        S
414         1306       1  female  39.0      0      0  108.9000        C
415         1307       3    male  38.5      0      0    7.2500        S
416         1308       3    male  38.5      0      0    8.0500        S
417         1309       3    male  38.5      1      1   22.3583        C

[418 rows x 8 columns]


Here i make interpolation and fillna to Age and Fare to fill the missing values then i used fillna in Embarked to write in the NaN items unavailable. Steps: Interpolate both age and fare by interpolation method then fillna inside each one of them that fill the null values with the value of interpolation of each of them.

In [10]:
test_data['Sex'] = preprocessing.LabelEncoder().fit_transform(test_data['Sex'])
test_data['Embarked'] = preprocessing.LabelEncoder().fit_transform(test_data['Embarked'])
test_data[['Sex','Embarked']].info()
test_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Sex       418 non-null    int64
 1   Embarked  418 non-null    int64
dtypes: int64(2)
memory usage: 6.7 KB


PassengerId  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0          892       3    1  34.5      0      0   7.8292         1
1          893       3    0  47.0      1      0   7.0000         2
2          894       2    1  62.0      0      0   9.6875         1
3          895       3    1  27.0      0      0   8.6625         2
4          896       3    0  22.0      1      1  12.2875         2

Here i converted all letters to numbers by using labelencoder.fittransform to sex and Embarked.
Steps:Using the method labelencoder to sex and Embarked. Conclusion: the letters in each are transformed to numbers as shown above.

In [11]:
y = train_data["Survived"]
X = train_data.drop("Survived",axis=1)
x_train,x_validation,y_train,y_validation=train_test_split(X,y,test_size=0.25,random_state=0)

Here i dropped column Survived from X and entered X as input and y is the output then i did the validation using train test split. Steps: Putting test_size= 0.25 and random_state=0.

In [12]:
model = RandomForestClassifier(n_estimators=100, max_depth=5, random_state=1)

Then here i used the model RandomForest giving it the parameters needed

In [13]:
X_test = test_data
print(X_test)
model.fit(x_train, y_train)
Score=model.score(x_validation,y_validation)
print(Score)
predictions = model.predict(X_test)

     PassengerId  Pclass  Sex   Age  SibSp  Parch      Fare  Embarked
0            892       3    1  34.5      0      0    7.8292         1
1            893       3    0  47.0      1      0    7.0000         2
2            894       2    1  62.0      0      0    9.6875         1
3            895       3    1  27.0      0      0    8.6625         2
4            896       3    0  22.0      1      1   12.2875         2
..           ...     ...  ...   ...    ...    ...       ...       ...
413         1305       3    1  33.5      0      0    8.0500         2
414         1306       1    0  39.0      0      0  108.9000         0
415         1307       3    1  38.5      0      0    7.2500         2
416         1308       3    1  38.5      0      0    8.0500         2
417         1309       3    1  38.5      1      1   22.3583         0

[418 rows x 8 columns]
0.8379888268156425


Here i printed the score and the accuarcy of the prediction of the model giving the score validation in parameters.Steps:putting test file in X_test and give the model.train the x_tain and y_train Conclusion: then calculating the score and printing the accuracy of the predictions.

In [14]:
output = pd.DataFrame({'PassengerId': test_data.PassengerId, 'Survived': predictions})
output.to_csv('My_Submission.csv', index=False)
output.head()
print(output)

     PassengerId  Survived
0            892         0
1            893         0
2            894         0
3            895         0
4            896         0
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]


Here i print the final output in submission file that shows the predicition of the model that shows the passengerId and the survived passengers. Steps: Giving the things that will appear in the output PassengerId and Survived passenger then translate them to file.csv called My_Submission.csv. conclusion: to see the output as shown above.